### PR TITLE
feat: gh action verification

### PR DESCRIPTION
**Pull request by jorgenbuilder**
- adds a gh action to allow for easy verification of the balance checker.
- adds a warning to the verification script when run on an operating system other than macos, with instructions on how to verify using the gh action.
- updates the readme to encourage use of the gh action.

### CODE REVIEW

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Verify Black Hole Canister
+on:
+  - push
+  - workflow_dispatch
+jobs:
+  verify:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (q-uint) [inline .github/workflows/verify.yml]: NIT: use `v3`.

**Review by q-uint — APPROVED** [review]

File: node/os.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import os from "os";
+
+export function verifyOS(): void {
+  const isMac = os.platform() === "darwin";
+  if (!isMac)
+    throw new Error(
+      "Verification must be performed on macos! The easiest way to do this is with a github action. See https://github.com/CycleOperators/BalanceCheckerVerification for github action instructions. This is due to differences in build environments resulting in different wasm hashes."
```
Comment (ByronBecker) [inline node/os.ts:7]:
```suggestion
      "To verify the controllers and hash of the blackholed CycleOps Balance Checker canister from a non-macOS operating system, use the following GitHub action. See https://github.com/CycleOperators/BalanceCheckerVerification for github action instructions.
      
      Currently, local verification is only supported on macOS! This is due to differences in build environments resulting in different wasm hashes.     
```

Comment (ByronBecker) [reply node/os.ts:7]: Call to action first with link for easier DevX (who reads error messages these days 😅 )

File: readme.md [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
+1. Fork this repository.
+2. Navigate to the actions tab of your new repo.
+3. Select the "Verify Black Hole Canister" action on the left.
+4. Dispatch the action.
+
+You can view the output of this action to confirm that 1) the canister has no controllers, 2) the mainnet wasm matches the contents of this repository. The final step to be 100% confident that this canister can do no harm is to audit the source code of this repo.
+
+## Alternative: Verify on your local machine
+
+_NOTE: wasm hash will vary depending on operating system. You must use macos to verify the wasm hash._
```
Comment (ByronBecker) [inline readme.md]:
```suggestion
_NOTE: Currently, this method is only supported for macOS, as the wasm hash generated will vary depending on operating system.
```

File: readme.md [hunk 4]
```diff
@@ -1,29 +1,47 @@
...
+
 To verify the CycleOps balance checker canister's black hole status from your local machine:
 
 1. Clone this repository
 2. Install version 0.13.1 of dfx (the version listed in the `dfx.json` file) with `DFX_VERSION=0.13.1 sh -ci "$(curl -fsSL https://internetcomputer.org/install.sh)"`. [dfx](https://internetcomputer.org/docs/current/references/cli-reference/dfx-parent/) is the DFINITY command-line execution environment for creating, deploying, and managing the dapps you develop for the IC.
 3. Install vessel from (https://github.com/dfinity/vessel). Vessel is used to fix the version of the [motoko-base](https://github.com/dfinity/motoko-base) library used in the project to 0.8.4.
-3. The following command:
+4. The following command:
```
Comment (ByronBecker) [inline readme.md:32]: Thanks 🤦

**Review by ByronBecker — CHANGES_REQUESTED** [review]
This is awesome work Jorgen! Thanks for getting this up and running.

A few small changes in terms of the error message wording around macOS compatibility

File: node/os.ts (hunk 2 above)
Comment (jorgenbuilder) [reply node/os.ts:7]: This seems like a longer way to say the same thing to me.

File: readme.md (hunk 3 above)
Comment (jorgenbuilder) [reply readme.md]: Looks good! Committing.

File: readme.md (hunk 4 above)
Comment (jorgenbuilder) [reply readme.md:32]: My linter did it! 😛

**Pull request merged**